inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/sink/JdbcSink.java

```diff
@@ -0,0 +1,164 @@
+import static org.apache.inlong.audit.config.ConfigConstants.KEY_SOURCE_DB_SINK_INTERVAL;
+import static org.apache.inlong.audit.config.ConfigConstants.PREP_STMT_CACHE_SIZE;
+import static org.apache.inlong.audit.config.ConfigConstants.PREP_STMT_CACHE_SQL_LIMIT;
+
+/**
+ * Jdbc sink
+ */
+public class JdbcSink implements AutoCloseable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcSink.class);
+    private final ScheduledExecutorService sinkTimer = Executors.newSingleThreadScheduledExecutor();
+    private final DataQueue dataQueue;
+    private final int insertBatch;
+    private final int pullTimeOut;
+    private final SinkConfig sinkConfig;
+    private DataSource dataSource;
+
+    public JdbcSink(DataQueue dataQueue, SinkConfig sinkConfig) {
+        this.dataQueue = dataQueue;
+        this.sinkConfig = sinkConfig;
+
+        insertBatch = Configuration.getInstance().get(KEY_SOURCE_DB_SINK_BATCH,
+                DEFAULT_SOURCE_DB_SINK_BATCH);
+
+        pullTimeOut = Configuration.getInstance().get(KEY_QUEUE_PULL_TIMEOUT,
+                DEFAULT_QUEUE_PULL_TIMEOUT);
+    }
+
+    /**
+     * start
+     */
+    public void start() {
+        createDataSource();
+
+        sinkTimer.scheduleWithFixedDelay(this::process,
+                0,
+                Configuration.getInstance().get(KEY_SOURCE_DB_SINK_INTERVAL,
+                        DEFAULT_SOURCE_DB_SINK_INTERVAL),
+                TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Process
+     */
+    private void process() {
+        try (Connection connection = dataSource.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(sinkConfig.getInsertSql())) {
+            if (connection.isClosed()) {
+                createDataSource();
+            }
+            int counter = 0;
+            StatData data = dataQueue.pull(pullTimeOut, TimeUnit.MICROSECONDS);
+            while (data != null) {
+                preparedStatement.setString(1, data.getLogTs());
+                preparedStatement.setString(2, data.getInlongGroupId());
+                preparedStatement.setString(3, data.getInlongStreamId());
+                preparedStatement.setString(4, data.getAuditId());
+                preparedStatement.setString(5, data.getAuditTag());
+                preparedStatement.setLong(6, data.getCount());
+                preparedStatement.setLong(7, data.getSize());
+                preparedStatement.setLong(8, data.getDelay());
+                preparedStatement.addBatch();
+
+                if (++counter >= insertBatch) {
+                    preparedStatement.executeBatch();
+                    preparedStatement.clearBatch();
+                    counter = 0;
+                }
+                data = dataQueue.pull(pullTimeOut, TimeUnit.MICROSECONDS);
+            }
+            if (counter > 0) {
+                preparedStatement.executeBatch();
```

Comment:
If the connection is disconnected, will there be a problem here

Reply:
Each cycle uses the latest data to overwrite the old data, so even if the connection is disconnected, it will not be affected.